editoast/src/generated_data/error/catenaries.rs

```diff
@@ -59,10 +67,51 @@ pub fn check_catenary_track_ranges(
     infra_errors
 }
 
+/// Checks that speed sections overlapping
+pub fn check_overlapping(infra_cache: &InfraCache, _: &Graph) -> Vec<InfraError> {
+    let mut overlapping_sc = HashSet::new();
+    // Key: (Track, Tag, Direction)
+    //   - Tag is None for the default speed limit and Some for the tagged ones
+    // Value: RangeMap<Range, SpeedSectionId>
+    let mut range_maps: HashMap<(String, Option<String>, Direction), RangeMap<u64, String>> =
+        Default::default();
+
+    // Iterate over all the speed sections insure we don't report duplicated errors
+    for catenary in infra_cache.catenaries().values() {
```

Comment:
That's just a personal preference bu there's 4 nested for loops, maybe put some in another function?